src/sentry/snuba/metrics/datasource.py

```diff
@@ -294,11 +289,6 @@ def get_tag_values(
                 metric_id = row["metric_id"]
                 tag_values[metric_id].append(value_id)
 
-        # If we are trying to find the tag values for only one metric name, then no need to query
```

Comment:
This loop would be exited prematurely even if a derived metric had constituent metrics that span multiple entities